app/components/pay-page/choose-membership-plan-modal/plan-card.hbs

```diff
@@ -0,0 +1,110 @@
+<div
+  class="flex items-start justify-between border
+    {{if @isSelected 'border-teal-500' 'border-gray-200 hover:border-gray-300'}}
+    rounded p-3
+    {{if @isSelected 'bg-teal-50' 'bg-white hover:bg-gray-50'}}"
+  role="button"
+  {{on "click" this.handleClick}}
+  data-test-modal-plan-card
+  ...attributes
+>
+  <div class="flex items-start gap-2">
+    <div
+      class="flex-shrink-0 flex items-center justify-center w-4 h-4 border rounded-full mt-0.5 {{if @isSelected 'border-teal-500' 'border-gray-400'}}"
+    >
+      {{#if @isSelected}}
+        <div class="w-2.5 h-2.5 bg-teal-500 rounded-full"></div>
+      {{/if}}
+    </div>
+
+    <div>
+      <div class="flex items-center gap-x-1.5">
+        <span class="text-gray-600 font-bold text-sm">
+          {{@plan.title}}
+        </span>
+        {{#if (eq @plan.id "yearly")}}
+          <span class="inline-block bg-teal-500 text-white font-bold rounded px-1.5 py-1 text-xxs uppercase">best value</span>
+        {{/if}}
+      </div>
+
+      <div class="prose prose-xs prose-compact mt-1 text-gray-500">
+        {{#if @plan.validityInMonths}}
+          <p>
+            Access to all membership benefits for
+            {{@plan.validityInMonths}}
+            {{if (eq @plan.validityInMonths 1) "month" "months"}}
+          </p>
+        {{else}}
+          <p>
+            Access to all membership benefits forever
+          </p>
+        {{/if}}
+      </div>
+
+      {{#if this.discountedPriceInDollars}}
+        <div class="prose prose-xs prose-compact text-gray-500">
+          <ul class="mt-0">
+            {{#if @regionalDiscount}}
+              <li data-test-modal-regional-discount-notice>
+                <b class="text-teal-500">{{@regionalDiscount.percentOff}}% off</b>
+                —
+                {{@regionalDiscount.countryName}}
+                discount
+              </li>
+            {{/if}}
+            {{#if (eq @plan.id "yearly")}}
```

Comment:
There's a bit of duplication of logic here, we're maintaining the fact that certain discounts only apply to the yearly plan both here and in the place where we calculate the discounted price. 

Idea to clean this up: 

- Add `get applicableDiscounts()`, which just returns [](PromotionalDiscount | RegionalDiscount)
- Use that in the `discountedPriceInDollars` calculation, and also here use it as `{{#each this.applicableDiscounts}}`.

This ensures that whatever is used in the calculation will be handled here too, without needing to add any extra logic in the template